src/main/java/org/dcsa/ebl/service/impl/ShippingInstructionServiceImpl.java

```diff
@@ -72,80 +76,212 @@
+                        activeReeferSettingsRepository.deleteByShipmentEquipmentID(
+                            cargoItems.getShipmentEquipmentID()))
+                    .thenReturn(cargoItems))
+        .flatMap(
+            cargoItem ->
+                shipmentEquipmentRepository
+                    .findShipmentEquipmentByShipmentID(cargoItem.getShipmentEquipmentID())
+                    .flatMap(
+                        shipmentEquipment ->
+                            equipmentRepository
+                                .deleteAllByEquipmentReference(
+                                    shipmentEquipment.getEquipmentReference())
+                                .thenReturn(shipmentEquipment))
+                    .flatMap(
+                        shipmentEquipment ->
+                            shipmentEquipmentRepository
+                                .deleteShipmentEquipmentByShipmentID(
+                                    shipmentEquipment.getShipmentID())
+                                .thenReturn(new ShipmentEquipmentTO()))
+                    .thenReturn(cargoItem))
+        .flatMap(
+            cargoItem ->
+                cargoLineItemRepository
+                    .deleteByCargoItemID(cargoItem.getId())
+                    .thenReturn(cargoItem))
+        .flatMap(
+            cargoItem -> cargoItemRepository.deleteById(cargoItem.getId()).thenReturn(cargoItem))
+        .collectList()
+        .flatMap(ignored -> insertShipmentEquipmentTOs(shipmentEquipments, shippingInstructionTO));
+  }
+
+  private Mono<List<ShipmentEquipmentTO>> insertShipmentEquipmentTOs(
+      List<ShipmentEquipmentTO> shipmentEquipments, ShippingInstructionTO shippingInstructionTO) {
+    if (shipmentEquipments == null) return Mono.empty();
+    String carrierBookingReference = getCarrierBookingReference(shippingInstructionTO);
+    // TODO: we have a known bug here that needs to be addressed.
+    //  carrierBookingReference can differ from each cargoItem.
+    return shipmentRepository
+        .findByCarrierBookingReference(carrierBookingReference)
+        .switchIfEmpty(
+            Mono.error(
+                ConcreteRequestErrorMessageException.invalidParameter(
+                    "No shipment found with carrierBookingReference: " + carrierBookingReference)))
+        .flatMap(
+            x ->
+                shipmentEquipmentService.createShipmentEquipment(
+                    x.getShipmentID(),
+                    shippingInstructionTO.getShippingInstructionID(),
+                    shipmentEquipments));
+  }
+
+  private Mono<LocationTO> insertLocationTO(LocationTO placeOfIssue, String shippingInstructionID) {
+    if (placeOfIssue == null) return Mono.empty();
+    return locationService.createLocationByTO(
+        placeOfIssue,
+        poi -> shippingInstructionRepository.setPlaceOfIssueFor(poi, shippingInstructionID));
+  }
+
+  private Mono<List<DocumentPartyTO>> resolveDocumentPartiesForShippingInstructionID(
```

Comment:
This method should be moved to event_core DocumentPartyService (and DocumentPartyServiceImpl)

Reply:
done